### PR TITLE
Feature/rm deprecate security explore

### PR DESCRIPTION
**Pull request by gsps1**
removing deprecated API in cdap-security/spi and explore modules.

### CODE REVIEW

File: cdap-security/src/test/java/co/cask/cdap/security/authorization/DummyAuthEnforce.java [hunk 1]
```diff
@@ -45,14 +45,6 @@ public void testSingleAction(@Name("namespaceId") NamespaceId namespaceId) throw
       throw new EnforceNotCalledException();
     }
 
-    @Deprecated // tests that the presence of other annotations does not affect class rewrite
```
Comment (gsps1) [inline cdap-security/src/test/java/co/cask/cdap/security/authorization/DummyAuthEnforce.java:48]: @rohitsinha54 Can you confirm if this is safe to be removed.

**Review by anwar6953 — APPROVED** [review]

File: cdap-security/src/test/java/co/cask/cdap/security/authorization/DummyAuthEnforce.java (hunk 1 above)
Comment (gsps1) [reply cdap-security/src/test/java/co/cask/cdap/security/authorization/DummyAuthEnforce.java:48]: checked with @rohitsinha54, This is just to test presence of additional annotations, so clarified the note with additional information.

File: cdap-security/src/test/java/co/cask/cdap/security/authorization/DummyAuthEnforce.java [hunk 2]
```diff
@@ -45,6 +45,14 @@ public void testSingleAction(@Name("namespaceId") NamespaceId namespaceId) throw
       throw new EnforceNotCalledException();
     }
 
+    @Deprecated // NOTE : Not deprecated, tests that the presence of other annotations does not affect class rewrite
```
Comment (anwar6953) [inline cdap-security/src/test/java/co/cask/cdap/security/authorization/DummyAuthEnforce.java]: can we use something like `@VisibleForTesting`, then?

Comment (gsps1) [reply cdap-security/src/test/java/co/cask/cdap/security/authorization/DummyAuthEnforce.java]: yeah, thats a better annotation to use for testing. i have updated it.

**Comment by gsps1** [comment]
build green https://builds.cask.co/browse/CDAP-DUT6197-5, squashed related last 4 commits, merging.

**Pull request merged**